### ISSUE TITLE
RgbInvoice and RgbTransport clarifications

### ISSUE THREAD

**Issue by zoedberg** [RGB-WG/rgb-std#34]
Hi @dr-orlovsky!
I'm working on a PR to change the `transport` field of the `RgbInvoice` from `RgbTransport` to `Vec<RgbTransport>`, since, as we discussed in a previous call, this would allow use cases where a user can provide multiple consignment endpoints, so that the sender can fallback to other transports if the first specified is not working (e.g. a storm service being temporarily down).
While working on this I've noticed the `RgbTransport` enum is currently defined as:

```
pub enum RgbTransport {
    #[display("rgb-rpc{tls}://host/")]
    JsonRpc { tls: bool, host: String },
    #[display("rgb+http{tls}://host/")]
    RestHttp { tls: bool, host: String },
    #[display("rgb+ws{tls}://host/")]
    WebSockets { tls: bool, host: String },
    #[display("rgb+storm://_/")]
    Storm {/* todo */},
    #[display("rgb:")]
    UnspecifiedMeans,
}
```

I don't think that `JsonRpc` (and similar variants as `RestHttp` and `WebSockets`) needs the `tls` boolean field since IMO it's more readable if we just encode the use or not-use of it in the host string (`http://host` when no tls and `https://host` when using tls).

Moreover I'm not sure how we should join multiple transports in the URI. I've tried joining the transports with a `+` but then the URI parser library (`fluent_uri`) fails parsing the URI schema, so before looking for a way to fix this I want to make sure what is in your opinion the best way to join multiple transport schemas in the URI. Thank you

**Comment by dr-orlovsky** [comment]
> I don't think that JsonRpc (and similar variants as RestHttp and WebSockets) needs the tls boolean field since IMO it's more readable if we just encode the use or not-use of it in the host string (http://host when no tls and https://host when using tls).

My assumption was that we will skip `http`/`https` parts and just provide a host information. Otherwise, (1) it would break URL integrity (adding more schemata will make most of libraries to fail, as you have seen); (2) no need to provide transport information since RgbRpc can work only with HTTP.

> Moreover I'm not sure how we should join multiple transports in the URI. I've tried joining the transports with a + but then the URI parser library (fluent_uri) fails parsing the URI schema

Hm, need to try different options.
If we use the same host address for different transports, we may just add `+` at the schema part, which would work (lie `rgb-rpc+storm://`. However, if the hosts are different - that what makes the story hard. Will try to play with the format of URL which may work for that.

**Comment by dr-orlovsky** [comment]
Closed wth #43
